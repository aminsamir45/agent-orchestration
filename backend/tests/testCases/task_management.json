{
  "name": "task_management",
  "description": "I need an agent system that helps manage tasks and projects. The system should have components that can:\n\n1. Task Collector Agent: This agent should collect tasks from various sources including emails, Slack messages, and manual input. It should extract task details like deadlines, priorities, and descriptions.\n\n2. Task Organizer Agent: This agent should categorize and prioritize tasks based on deadlines, importance, and dependencies between tasks. It should maintain a structured database of all tasks.\n\n3. Reminder Agent: This agent should send timely notifications about upcoming deadlines and important tasks that need attention. It should be able to send notifications through multiple channels.\n\n4. Progress Tracking Agent: This agent should monitor task completion status and generate reports on progress. It should identify bottlenecks and suggest optimizations.\n\n5. Resource Allocation Agent: This agent should analyze resource requirements for tasks and suggest optimal allocation of team members based on skills and availability.\n\nThe agents should work together in a coordinated way where the Task Collector feeds information to the Task Organizer, which then provides data to the Reminder Agent, Progress Tracking Agent, and Resource Allocation Agent. The system should have a central database that all agents can access and update.\n\nFor integrations, the system needs to connect with:\n- Email services (Gmail, Outlook)\n- Calendar applications\n- Slack\n- Project management tools like Jira and Trello\n- Time tracking software\n\nThe system should be able to handle multiple projects simultaneously and provide both detailed views of individual tasks and high-level overviews of project status."
} 