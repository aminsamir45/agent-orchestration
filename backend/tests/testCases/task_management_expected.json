{
  "summary": "A multi-agent task and project management system with capabilities for task collection, organization, reminders, progress tracking, and resource allocation.",
  "agents": [
    {
      "name": "Task Collector Agent",
      "role": "Data Gathering",
      "description": "Collects tasks from various sources including emails, Slack messages, and manual input.",
      "required_capabilities": ["email integration", "slack integration", "data extraction"]
    },
    {
      "name": "Task Organizer Agent",
      "role": "Data Processing",
      "description": "Categorizes and prioritizes tasks based on deadlines, importance, and dependencies between tasks.",
      "required_capabilities": ["database management", "priority algorithm", "categorization"]
    },
    {
      "name": "Reminder Agent",
      "role": "Notification",
      "description": "Sends timely notifications about upcoming deadlines and important tasks.",
      "required_capabilities": ["scheduling", "multi-channel notifications"]
    },
    {
      "name": "Progress Tracking Agent",
      "role": "Monitoring",
      "description": "Monitors task completion status and generates reports on progress.",
      "required_capabilities": ["data visualization", "analytics", "bottleneck detection"]
    },
    {
      "name": "Resource Allocation Agent",
      "role": "Resource Management",
      "description": "Analyzes resource requirements and suggests optimal allocation of team members.",
      "required_capabilities": ["skill matching", "availability tracking", "workload balancing"]
    }
  ],
  "tools": [
    {
      "name": "Email Integration",
      "purpose": "Connect with email services like Gmail and Outlook",
      "usedBy": ["Task Collector Agent"]
    },
    {
      "name": "Calendar Integration",
      "purpose": "Sync with calendar applications for scheduling",
      "usedBy": ["Reminder Agent", "Resource Allocation Agent"]
    },
    {
      "name": "Slack Connector",
      "purpose": "Interface with Slack for messages and notifications",
      "usedBy": ["Task Collector Agent", "Reminder Agent"]
    },
    {
      "name": "Project Management API",
      "purpose": "Connect with tools like Jira and Trello",
      "usedBy": ["Task Organizer Agent", "Progress Tracking Agent"]
    },
    {
      "name": "Time Tracking Integration",
      "purpose": "Interface with time tracking software",
      "usedBy": ["Progress Tracking Agent", "Resource Allocation Agent"]
    },
    {
      "name": "Database System",
      "purpose": "Central storage for all task and project data",
      "usedBy": ["Task Organizer Agent", "Progress Tracking Agent", "Resource Allocation Agent"]
    }
  ],
  "relationships": [
    {
      "source": "Task Collector Agent",
      "target": "Task Organizer Agent",
      "description": "Data flow from collection to organization",
      "dataFlow": "Raw task data"
    },
    {
      "source": "Task Organizer Agent",
      "target": "Reminder Agent",
      "description": "Providing organized task data for notifications",
      "dataFlow": "Task deadlines and priorities"
    },
    {
      "source": "Task Organizer Agent",
      "target": "Progress Tracking Agent",
      "description": "Sharing task structure for tracking",
      "dataFlow": "Organized task data with dependencies"
    },
    {
      "source": "Task Organizer Agent",
      "target": "Resource Allocation Agent",
      "description": "Providing task requirements for resource planning",
      "dataFlow": "Task details and resource requirements"
    }
  ],
  "orchestrationPattern": {
    "type": "Hierarchical",
    "justification": "The Task Organizer acts as a central hub that receives data from the Collector and distributes to other agents"
  },
  "constraints": [
    "Must handle multiple projects simultaneously",
    "Needs to provide both detailed and high-level views",
    "Should support various third-party integrations"
  ]
} 